project-docs/javascript-linter-spec.md

```diff
@@ -0,0 +1,163 @@
+
+#### prose.\*
+
+This ingredient enables writers to include extra sections, but ensuring that they always appear in the same place in the page. To satisfy this ingredient: if the page contains any `<h2>` headings that are not expected by the recipe, then they must appear at the point in the page that the `prose.*` ingredient is listed in the recipe.
+
+### Data ingredients
+
+Data ingredients typically impose more detailed requirements on pages than prose ingredients.
+
+#### data.interactive_example?
+
+This is an optional data ingredient. To satisfy the ingredient, if the page contains a call to the `{{EmbedInteractiveExample}}` macro, then the call must appear in the place specified by the recipe.
+
+#### data.class_constructor
+
+To satisfy this ingredient, a page must meet the following requirements:
+
+- A section demarcated by `H2#Constructor` that contains only a `<dl>` element.
+- The `<dl>` must contain a single `<dt><dd>` pair.
+- The `<dt>` must contain either:
+  - only a single `<code>` element, that contains only a single `<a>` element
+  - only a call to the `{{jsxref}}` macro.
+
+#### data.constructor_properties?
+
+This is an optional ingredient. If a page contains a section demarcated by `H2#Constructor_properties`, then it must meet the following requirements:
+
+- Must contain only a `<dl>` element.
+- The `<dl>` must contain one or more `<dt><dd>` pairs.
+- The `<dt>` must contain either:
+  - only a single `<code>` element, that contains only a single `<a>` element
+  - only a call to the `{{jsxref}}` macro.
+- `<dt><dd>` pairs must be ordered by the alphabetical order of the `<dt><code><a>` text content, or by the display name given in the `{{jsxref}}` macro call.
+
+#### data.static_properties?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Static_properties`.
+
+#### data.static_methods?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Static_methods`.
+
+#### data.instance_properties?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Instance_properties`.
+
+#### data.instance_methods?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Instance_methods`.
+
+#### data.examples
+
+To satisfy this ingredient a page must have a section demarcated by `H2#Examples`. It must contain one or more sections demarcated by an H3. Each of these sections represents a single example.
+
+Each of these examples may include "code sections". These are marked up as follows:
+
+- HTML code section: `<pre class="brush: html language-html"><code class=" language-html">`
```

Comment:
Are these spaces here and for CSS and JS on purpose?

```suggestion
- HTML code section: `<pre class="brush: html language-html"><code class="language-html">`
```

Reply:
This is what the markup currently looks like, I just copy/pasted it. But you're right, there's no reason to have it here that I can see.